### DISCUSSION TITLE
Is there something like renovatebot core?

### DISCUSSION THREAD

**Discussion by rassie** [renovatebot/renovate#8311] [Suggest an Idea]
I have an idea for a bot. It should work exactly like renovatebot, i.e. look into repositories, do some processing, open issues, keep track of rejected proposals etc., but the focus would be checking for company-specific project structure, i.e. nothing to do with renovatebot's mission of keeping dependencies current.

Is there any way I could leverage renovatebot's existing algorithms to implement new functionality on top? Something like renovatebot Core would be very nice :)

**Comment by rarkins** [comment]
No, this repo is as core as it gets. The project is a few years old but we've regularly refactored and reorganized our own structure and interfaces so it still hasn't reached a point where I'd want to break it into modules and commit to not changing them regularly.

That said, we have full TypeScript coverage so if you wrote your project in TS and directly required certain aspects of the bot code then you can still get compile-time warnings if any interfaces change.

**Comment by rassie** [comment]
OK, thanks, I will look into it!
